### DISCUSSION TITLE
Removing an existing handler

### DISCUSSION THREAD

**Discussion by me21** [zauberzeug/nicegui#3440] [Q&A] [answered]
### Question

Hello,

Suppose I have a button with `on_click` handler. Is it possible to remove the current handler from the button and assign another? If I understand correctly, calling `btn.on_click(my_new_handler)` appends it to the handlers list, keeping the old handler as well.

**Comment by python-and-novella** [comment]
```python3
from nicegui import ui

ub = ui.button("handler")
ub.on("click", ui.notify)
ub._event_listeners.clear()
ub.on("click", lambda: print(1))
ub.on("click", lambda: print(2))

ui.run()
```

**Reply by falkoschindler** [reply]
This might _kind of_ work, but introduces a memory leak: The client will send events for every event listener ever created. So if you replace the listeners again and again, the client will send loads of events which the server doesn't need anymore.

**Comment by falkoschindler** [comment, marked as answer]
Currently there is no way to cleanly unregister event listeners. We usually recommend subscribing to the event once and letting the event handler decide what to do, e.g. printing "1", printing "2", notifying etc.

**Comment by Mohameme7** [comment]
no fixes for that yet or any additions?

**Reply by evnchn** [reply]
Not any so far IIRC.

If you have a strong demand for it, PRs are welcome!
